opencog/reducture/rules/logical-rules.scm

```diff
@@ -0,0 +1,234 @@
+
+(DefineLink
+ and-identity-true
+   (BindLink
+     (Glob "$op")
+     (Present
+     (AndLink
+       (Glob "$op")
+       (True)))
+
+     (ReducedTo
+     (AndLink
+     (Glob "$op")
+     (True))
+     (Glob "$op"))
+ ))
+
+;; or(true X)->true
+
+(DefineLink
+ or-identity-true
+   (BindLink
+     (Glob "$op")
+     (Present
+     (OrLink
+       (Glob "$op")
+       (True)))
+
+     (ReducedTo
+     (OrLink
+     (Glob "$op")
+     (True))
+     (True)
+     )
+ ))
+
+;; and(false X)->false
+
+(DefineLink
+ and-identity-false
+   (BindLink
+     (Glob "$op")
+     (Present
+     (AndLink
+       (Glob "$op")
+       (False)))
+
+     (ReducedTo
+     (AndLink
+     (Glob "$op")
+     (False))
+     (False))
+ ))
+
+;; or(false X)->X
+(DefineLink
+ or-identity-false
+   (BindLink
+     (Glob "$op")
+     (Present
+     (OrLink
+       (Glob "$op")
+       (False)))
+
+     (ReducedTo
+     (OrLink
+     (Glob "$op")
+     (False))
+     (Glob "$op")
+     )
+ ))
+
+;; or(X)->X
+
+(DefineLink
+ or-identity
+   (BindLink
+     (Glob "$op")
+     (Present
+     (OrLink
+       (Glob "$op")))
+
+     (ReducedTo
+     (OrLink
+     (Glob "$op"))
+     (Glob "$op"))
+ ))
+
+;; and(or X) -> and(X)
+
+(DefineLink
+  and-or-identity
+   (BindLink
+     (Glob "$op")
+     (Present
+     (AndLink
+       (OrLink
+         (Glob "$op"))))
+
+     (ReducedTo
+     (AndLink
+      (OrLink
+        (Glob "$op")))
+
+     (AndLink
+      (Glob "$op")) )
+ ))
+
+
+;; or(and X) -> or(X)
+
+(DefineLink
+  or-and-identity
+   (BindLink
+     (Glob "$op")
+     (Present
+     (OrLink
+       (AndLink
+         (Glob "$op"))))
+
+     (ReducedTo
+     (OrLink
+      (AndLink
+        (Glob "$op")))
+
+     (OrLink
+      (Glob "$op")) )
+ ))
+
+;; !!a->a,
+
+ (DefineLink
+   double-not
+    (BindLink
+      (Variable "x")
+      (Present
+      (NotLink
+        (NotLink
+          (Variable "x"))))
+
+      (ReducedTo
+      (NotLink
+       (NotLink
+         (Variable "x")))
+       (Variable "x"))
+  ))
+;; !(a&&b)->(!a||!b)
+
+ (DefineLink
+   distribuitive-and
+    (BindLink
+    (VariableList
+      (Variable "a")
+      (Variable "b"))
+      (Present
+      (NotLink
+        (AndLink
+          (Variable "a")
+          (Variable "b")
+          )))
+
+      (ReducedTo
+      (NotLink
+       (AndLink
+         (Variable "a")
+         (Variable "b")))
+
+       (OrLink
+         (NotLink
+         (Variable "a"))
+         (NotLink
+         (Variable "b")))
+       )
+))
+
+;; !(a||b)->(!a&&!b)
+
+(DefineLink
+   distribuitive-or
+    (BindLink
+    (VariableList
+      (Variable "a")
+      (Variable "b"))
+      (Present
+      (NotLink
+        (OrLink
+          (Variable "a")
+          (Variable "b")
+          )))
+
+      (ReducedTo
+      (NotLink
+       (AndLink
+         (Variable "a")
+         (Variable "b")))
+
+       (AndLink
+         (NotLink
+         (Variable "a"))
+         (NotLink
+         (Variable "b")))
+       )
+))
```

Comment:
Missing new line.

Reply:
Can you point out to me where exactly i am missing it?

Reply:
Eman at the end of the file you should have a new line.

Reply:
@kasimebrahim  oh yeah I have seen that. Btw just a side note @ngeiswei  @kasimebrahim  the HAVE_URE tag on cmake is not actually finding my URE.